src/pages/dataTableUtilities/actions/cellActions.js

```diff
@@ -17,6 +17,33 @@ import { openModal, closeModal } from './pageActions';
  */
 export const refreshRow = rowKey => ({ type: ACTIONS.REFRESH_ROW, payload: { rowKey } });
 
+/**
+ * Edits a rows underlying `batch` field.
+ *
+ * @param {Date}    value       The new batch name value.
+ * @param {String}  rowKey      Key of the row to edit.
+ * @param {String}  objectType  Type of object to edit i.e. 'TransactionBatch'
+ */
+export const editBatchName = (value, rowKey, objectType) => (dispatch, getState) => {
+  const { data, keyExtractor } = getState();
+
+  const objectToEdit = data.find(row => keyExtractor(row) === rowKey);
+
+  const { batch } = objectToEdit;
+
+  if (value !== batch) {
+    UIDatabase.write(() => UIDatabase.update(objectType, { ...objectToEdit, batch: value }));
+
+    dispatch(refreshRow(rowKey));
+  }
+};
+
+/**
+ * Wrapper around editBatchName for StocktakeBatches
+ */
+export const editStocktakeBatchName = (value, rowKey) =>
```

Comment:
Seems like a bit of a [word here] of realm to not have a `[someRealmObject].type` field. I would like to have a `dataTypes/constants` folder which defines constants for every data type such that using `UIDatabase.objects("StocktakeBatch")` wasn't so yuck -> `UIDatabase.objects(TYPES.stocktakeBatch)` seems nicer. A lot of code to change and would prefer to do it all the way rather than a half-baked attempt and a little out of scope for the refactor issue, I think?